db/model/Gdoc/htmlToEnriched.ts

```diff
@@ -86,82 +95,998 @@ export function htmlToSimpleTextBlock(html: string): EnrichedBlockSimpleText {
+                            listContent.content,
+                            isEnrichedTextBlock
+                        )
+                        const errors = listContent.errors
+                        if (otherChildren.length > 0)
+                            errors.push({
+                                name: "unexpected elements in list item",
+                                details: `Found ${otherChildren.length} elements`,
+                            })
+                        return {
+                            errors: errors,
+                            content: [
+                                {
+                                    type: "text",
+                                    value: textChildren.flatMap(
+                                        (child) => child.value
+                                    ),
+                                    parseErrors: [],
+                                },
+                            ],
+                        }
+                    }
+
+                    const listChildren = joinBlockParseResults(
+                        children.map(handleListChildren)
+                    )
+                    return {
+                        errors: listChildren.errors,
+                        content: [
+                            {
+                                type: "numbered-list",
+                                items: listChildren.content,
+                                parseErrors: [],
+                            },
+                        ],
+                    }
+                }
+            )
+            .with(
+                { tagName: P.union("svg", "table", "video") },
+                (): BlockParseResult<EnrichedBlockHtml> => {
+                    return {
+                        errors: [],
+                        content: [
+                            {
+                                type: "html",
+                                value: context.$.html(element) ?? "",
+                                parseErrors: [],
+                            },
+                        ],
+                    }
+                }
+            )
+            .otherwise(() => ({
+                errors: [
+                    {
+                        name: "unhandled html tag found",
+                        details: `Encountered the unhandled tag ${element.tagName}`,
+                    },
+                ],
+                content: [],
+            }))
+        return result
+    } else
+        return {
+            errors: [
+                {
+                    name: "unkown element tag",
+                    details: `type was ${element.type}`,
+                },
+            ],
+            content: [],
+        }
+}
+
+export function cheerioElementsToArchieML(
+    elements: CheerioElement[],
+    context: ParseContext
+): BlockParseResult<ArchieBlockOrWpComponent> {
+    if (!elements || !elements.length)
```

Comment:
I'm getting the following error while running the migrate script

````
Caught an exception 2447
/home/owid/owid-grapher/itsJustJavascript/db/model/Gdoc/htmlToEnriched.js:713
    if (!elements.length)
                  ^
TypeError: Cannot read properties of undefined (reading 'length')
````

Adding a "?" fixes it (`if (!elements?.length)`) but maybe it's worth checking why `elements` is undefined for this article

Reply:
ah sorry, I'm running an old commit